### PR TITLE
Cleanup resources when initializing error in integration

### PR DESCRIPTION
**Pull request by kerthcet**
#### What type of PR is this?
/kind cleanup
/sig testing
/sig scheduling 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kerthcet** [comment]
Would you mind to take a look @aojea

**Review by alculquicondor — COMMENTED** [review]
/approve

**Comment by kerthcet** [comment]
Hi @pohly can you also take a look as test approver?

File: test/integration/scheduler/plugins/plugins_test.go [hunk 1]
```diff
@@ -2556,6 +2556,8 @@ func initTestSchedulerForFrameworkTest(t *testing.T, testCtx *testutils.TestCont
 
 	if nodeCount > 0 {
 		if _, err := createAndWaitForNodesInCache(testCtx, "test-node", st.MakeNode(), nodeCount); err != nil {
+			// Make sure to cleanup the resources when initializing error.
+			testutils.CleanupTest(t, testCtx)
```
Comment (pohly) [inline test/integration/scheduler/plugins/plugins_test.go:2560]: It looks like every single `initTestSchedulerForFrameworkTest` is followed by `defer testutils.CleanupTest`.

If `initTestSchedulerForFrameworkTest` did a `t.Cleanup(func() { testutils.CleanupTest(t, testCtx) })` directly after setting up resources, then all other calls of `CleanupTest` could be removed, right?

Comment (kerthcet) [reply test/integration/scheduler/plugins/plugins_test.go:2560]: Based on current implementations, yes, the `CleanupTest()` will tear down all created resources in integration tests, but what I want to add later is keeping the apiserver still running, but tear down the scheduler only, then we can't call `CleanupTest`, instead, I'll add a new function.

So the key point here is sometimes we want to tear down the whole testing environment, but somethings just part of them, better to keep it as it-is, advices?

Comment (kerthcet) [reply test/integration/scheduler/plugins/plugins_test.go:2560]: The discussion came from here https://github.com/kubernetes/kubernetes/pull/114717#discussion_r1108762317

Comment (pohly) [reply test/integration/scheduler/plugins/plugins_test.go:2560]: With that goal in mind the proposed change makes sense.

/lgtm
/approve

**Pull request merged**